Disable all tests that require a leak sanitizer because
OpenBSD does not support it.

Disable symbolize_test, which fails to link on amd64 where ld.lld
defaults to --exec-only.

Index: absl/debugging/CMakeLists.txt
--- absl/debugging/CMakeLists.txt.orig
+++ absl/debugging/CMakeLists.txt
@@ -75,6 +75,7 @@ absl_cc_library(
   PUBLIC
 )
 
+if (NOT CMAKE_SYSTEM_NAME STREQUAL "OpenBSD")
 absl_cc_test(
   NAME
     symbolize_test
@@ -96,6 +97,7 @@ absl_cc_test(
     absl::strings
     GTest::gmock
 )
+endif()
 
 # Internal-only target, do not depend on directly.
 absl_cc_library(
@@ -206,6 +208,7 @@ absl_cc_test(
     GTest::gmock_main
 )
 
+if (NOT CMAKE_SYSTEM_NAME STREQUAL "OpenBSD")
 absl_cc_library(
   NAME
     leak_check
@@ -235,6 +238,7 @@ absl_cc_test(
     absl::base
     GTest::gmock_main
 )
+endif()
 
 # Internal-only target, do not depend on directly.
 absl_cc_library(
@@ -253,6 +257,7 @@ absl_cc_library(
   TESTONLY
 )
 
+if (NOT CMAKE_SYSTEM_NAME STREQUAL "OpenBSD")
 absl_cc_test(
   NAME
     stack_consumption_test
@@ -278,3 +283,4 @@ absl_cc_library(
     absl::leak_check
   PUBLIC
 )
+endif()
